### PR TITLE
feat(functions): add CloudEvents-based PubSub sample

### PR DESCRIPTION
**Pull request by ace-n**
@grayside LMK if I should add a second (Node.js-focused) reviewer.

### CODE REVIEW

**Review by JustinBeckwith — APPROVED** [review]
Can confirm this is JavaScript

File: functions/v2/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.zip
```
Comment (grayside) [inline functions/v2/.gitignore:1]: question: Why is this needed?
suggestion: Can it be added to the root gitignore? Scattered ignore files can be confusing.

File: functions/v2/test/index.test.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const assert = require('assert');
+const {request} = require('gaxios');
+const {exec} = require('child_process');
+const waitPort = require('wait-port');
+
+const startFF = async (target, signature, port) => {
```
Comment (grayside) [inline functions/v2/test/index.test.js:20]: suggestion: I tend to feel system tests using the function framework locally have a high overhead relative to their advantage over unit tests. Would kicking off v2 samples be a good moment to switch to Deploy tests?

**Review by grayside — APPROVED** [review]
No blocking feedback. Optional suggestions to use or file for follow-up at your discretion.

Comments attempting to follow conventionalcomments.org

File: functions/v2/.gitignore (hunk 1 above)
Comment (ace-n) [reply functions/v2/.gitignore:1]: Currently, the GCF V2 deployment process sometimes generates extra `.zip` files.

This is a "convenience feature" to prevent us accidentally including them in commits.

File: functions/v2/test/index.test.js (hunk 2 above)
Comment (ace-n) [reply functions/v2/test/index.test.js:20]: I believe the Functions Frameworks are directly integrated into [all versions of] GCF, so this will give us a medium-high fidelity test.

I agree with your sentiment, but I think we need to be more organized/deliberate about how we add deployment tests. (In particular, do we want to write deployment tests in every language - or have some common "translation" layer so we only need write them in one language [e.g. PHP, which already has deployment tests]?)
